### PR TITLE
Update for GStreamer repo reorg

### PR DESCRIPTION
**Pull request by hadess**
Take the opportunity of GStreamer mono repository reorganisation:
https://gstreamer.freedesktop.org/documentation/frequently-asked-questions/mono-repository.html
to build the whole of the GStreamer stack ourselves.

GStreamer usually releases its new point releases in October, the
Freedesktop Flatpak runtime in August. This means that our application
would get updates nearly a year after the fact, a pace somewhat too slow
for an application that relies so much on upstream bug fixes to work
properly and handle "edge cases" media files.

### CODE REVIEW

File: codecs/move-gst-plugins.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+mkdir -p /app/lib/codecs/lib/gstreamer-1.0
+export GST_PLUGIN_SYSTEM_PATH=/app/lib/gstreamer-1.0
+
+for i in /app/lib/gstreamer-1.0/*.so; do
+  SUITE=`gst-inspect-1.0 $i | grep "Source module" | sed 's,Source module,,' | tr -d [:blank:]`
+  if [ x$SUITE != x'gst-plugins-bad' ] &&
+     [ x$SUITE != x'gst-plugins-ugly' ] &&
+     [ x$SUITE != x'gst-libav' ]; then
```
Comment (wjt) [inline codecs/move-gst-plugins.sh:10]: Is this sense the right way round – only move `$i` if it is **not** in -bad / -ugly / -libav?

Comment (hadess) [reply codecs/move-gst-plugins.sh:10]: Definitely the wrong way around:
```
$ ls /app/lib/gstreamer-1.0/libgstlibav.so 
/app/lib/gstreamer-1.0/libgstlibav.so
```

**Comment by wjt** [comment]
Some stuff that could be `cleanup`ed:

./libexec/gnome-desktop-debug
./libexec/gnome-desktop-debug/gnome-rr-debug
./libexec/gnome-desktop-debug/test-desktop-thumbnail
./libexec/gnome-desktop-debug/test-idle-monitor
./libexec/gnome-desktop-debug/test-languages
./libexec/gnome-desktop-debug/test-pnp-ids
./libexec/gnome-desktop-debug/test-wall-clock
./libexec/gnome-desktop-debug/test-xkb-info
lib/gstreamer-1.0/include/gst/gl/gstglconfig.h
./lib/tracker-3.0/trackertestutils
./share/doc

And in `.Codecs`:

- `lib/libdvdnav.la` and other `*.la`

From a "patent-encumbered stuff is split into the `.Codecs` runtime" perspective this LGTM. Haven't actually run it though :

**Comment by hadess** [comment]
>  lib/gstreamer-1.0/include/gst/gl/gstglconfig.h

I couldn't figure out why this sodding file just wouldn't disappear, so I'm leaving it be.

**Pull request merged**